src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -1061,54 +1063,111 @@ void WindowImplX11::setVideoMode(const VideoMode& mode)
     if (mode == VideoMode::getDesktopMode())
         return;
 
-    // Check if the XRandR extension is present
-    int version;
-    if (!XQueryExtension(m_display, "RANDR", &version, &version, &version))
+    // Check if the XRandR extension
+    int xRandRMajor, xRandRMinor;
+    if (!checkXRandR(xRandRMajor, xRandRMinor))
```

Comment:
If we don't care about the actual version, but just the presence of the extension, then I'd remove the two parameters of `checkXRandR`. Having the function is fine, but the simpler it is the better.